### PR TITLE
Add multichannel support to RandomShadow (#1735)

### LINKED ISSUE #1735: [Feature Request] Add multichannel support to RandomShadow
RIght now we have:

```python
mask = np.zeros_like(img, dtype=np.uint8)
cv2.fillPoly(mask, vertices_list, (max_value, max_value, max_value))
```
in `add_shadow` function.

`cv2.fillPolly`, does not work with the number of channels > 4

=> 

1. if number of channels is larger than 4, create mask for 1 channel and replicate it to the desired number of channels.
2. check that everything works for grayscale image
3. tests
4. update docstrings in `add_shadow` and `RandomShadow`

### PR DESCRIPTION
**Pull request by JonasKlotz**
closes #1735

## Summary by Sourcery

Add multichannel support to the RandomShadow transformation by updating the add_shadow function to handle images with more than three channels. Adjust the mask creation process to ensure compatibility with multichannel images and update relevant documentation.

New Features:
- Add support for multichannel images in the RandomShadow transformation, allowing shadows to be applied to images with more than three channels.

Enhancements:
- Modify the add_shadow function to handle multichannel images by adjusting the mask creation process to match the number of channels in the input image.

Tests:
- Remove the RandomShadow transformation from the list of transformations in a specific test, possibly to update or replace the test with a new one that supports multichannel images.

### CODE REVIEW

File: albumentations/augmentations/functional.py [hunk 1]
```diff
@@ -664,20 +664,22 @@ def add_shadow(img: np.ndarray, vertices_list: list[np.ndarray]) -> np.ndarray:
...
+    # Create mask with shape (height, width, 1)
+    mask = np.zeros((img.shape[0], img.shape[1], 1), dtype=np.uint8)
+    cv2.fillPoly(mask, vertices_list, (max_value,))
+    # Duplicate the mask to have the same number of channels as the image
+    mask = np.repeat(mask, num_channels, axis=2)
 
-    # Apply shadow to the RGB channels directly
+    # Apply shadow to the channels directly
     # It could be tempting to convert to HLS and apply the shadow to the L channel, but it creates artifacts
     shadow_intensity = 0.5  # Adjust this value to control the shadow intensity
```
[the remark this answers is not in the document]
Comment (JonasKlotz) [reply albumentations/augmentations/functional.py:684]: Good idea, I've already proposed that!

File: albumentations/augmentations/transforms.py [hunk 2]
```diff
@@ -1131,7 +1131,9 @@ def get_transform_init_args(self) -> dict[str, Any]:
 
 
 class RandomShadow(ImageOnlyTransform):
-    """Simulates shadows for the image
+    """Simulates shadows for the image by reducing the brightness of the image in the shadow regions.
+
+    Works for both single and multi-channel images.
```
[the remark this answers is not in the document]
Comment (JonasKlotz) [reply albumentations/augmentations/transforms.py:1136]: Just states the same information again. The transformation is quite simple, thus too much documentation might not be necessary.

**Comment by JonasKlotz** [comment]
> The changes for multichannel support look good, but could you explain why the RandomShadow() line was removed from test_transforms.py? This seems counterintuitive given that we're adding functionality to RandomShadow.

The line was removed as the test tests the warning message when the input image is not RGB. As I made multichannel images possible the test would fail by definition.

**Comment by ternaus** [comment]
> > The changes for multichannel support look good, but could you explain why the RandomShadow() line was removed from test_transforms.py? This seems counterintuitive given that we're adding functionality to RandomShadow.
> 
> The line was removed as the test tests the warning message when the input image is not RGB. As I made multichannel images possible the test would fail by definition.

That's fine. Suggestions by Sourcery are optional. It may find some interesting optimization, but most of the time it is off and could be ignored.

**Comment by ternaus** [comment]
@JonasKlotz Thanks for the pull request!

**Pull request merged**